src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -5190,5 +5314,1050 @@ public void ToString_UserDefinedPrintMembers_Static()
                 Diagnostic(ErrorCode.ERR_StaticAPIInRecord, "PrintMembers").WithArguments("C1.PrintMembers(System.Text.StringBuilder)").WithLocation(4, 25)
                 );
         }
+
+        [Fact]
+        public void AmbigCtor_WithPropertyInitializer()
+        {
+            // Scenario causes ambiguous ctor for record class, but not record struct
+            var src = @"
+record struct R(R X)
+{
+    public R X { get; init; } = X;
```

Comment:
Is there also a test for when no property is explicitly declared?

Reply:
Added. Thanks

---
In reply to: [606014509](https://github.com/dotnet/roslyn/pull/52256#discussion_r606014509) [](ancestors = 606014509)